x/storage/keeper/msg_server.go

```diff
@@ -600,3 +589,131 @@ func (k msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParam
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+func (k msgServer) MigrateBucket(goCtx context.Context, msg *types.MsgMigrateBucket) (*types.MsgMigrateBucketResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	operator := sdk.MustAccAddressFromHex(msg.Operator)
+
+	bucketInfo, found := k.GetBucketInfo(ctx, msg.BucketName)
+	if !found {
+		return nil, types.ErrNoSuchBucket
+	}
+
+	if bucketInfo.BucketStatus == types.BUCKET_STATUS_MIGRATING {
+		return nil, types.ErrInvalidBucketStatus.Wrapf("The bucket already been migrating")
+
+	}
+	srcSP, found := k.spKeeper.GetStorageProviderByOperatorAddr(ctx, operator)
+	if !found {
+		return nil, sptypes.ErrStorageProviderNotFound
+	}
+
+	dstSP, found := k.spKeeper.GetStorageProvider(ctx, msg.DstPrimarySpId)
+	if !found {
+		return nil, sptypes.ErrStorageProviderNotFound.Wrapf("dst sp not found")
+	}
+
+	if srcSP.Status == sptypes.STATUS_IN_SERVICE || dstSP.Status == sptypes.STATUS_IN_SERVICE {
```

Comment:
The check `srcSP.Status == sptypes.STATUS_IN_SERVICE || dstSP.Status == sptypes.STATUS_IN_SERVICE` is not correct.

Reply:
yes. Fixed. Change it to 

```
	if srcSP.Status != sptypes.STATUS_IN_SERVICE && dstSP.Status != sptypes.STATUS_IN_SERVICE {
		return nil, sptypes.ErrStorageProviderNotInService.Wrapf(
			"origin SP status: %s, dst SP status: %s", srcSP.Status.String(), dstSP.Status.String())
	}
```

Reply:
`&&` -> `||`, if i understand it correctly ?

Reply:
yes